src/domain/event/EditEventPage.tsx

```diff
@@ -79,6 +91,41 @@ const EditEventPage: React.FC = () => {
         })
       );
 
+      const venueData = apolloClient.readQuery<VenueQuery>({
+        query: VenueDocument,
+        variables: { id: values.location },
+      });
+
+      const venueShouldBeUpdated = Boolean(
+        venueData?.venue &&
+          values.locationDescription !== initialValues.locationDescription
+      );
+      const newVenueShouldBeCreated = Boolean(
+        !venueData?.venue && values.locationDescription
+      );
+
+      // Update venue info
+      if (venueShouldBeUpdated) {
+        requests.push(
+          editVenue({
+            variables: getExistingVenuePayload({
+              formValues: values,
+              selectedLanguage,
+              venueData: venueData as VenueQuery,
+            }),
+          })
+        );
+      } else if (newVenueShouldBeCreated) {
+        requests.push(
+          createVenue({
+            variables: getNewVenuePayload({
+              selectedLanguage,
+              formValues: values,
+            }),
+          })
+        );
+      }
```

Comment:
You could have helper function to get create or edit request. This code is pretty much same as in CreateEventPage.

Reply:
Yeah absolutely